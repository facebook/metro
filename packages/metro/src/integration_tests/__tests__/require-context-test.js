/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @oncall react_native
 */

'use strict';

const Metro = require('../../..');
const execBundle = require('../execBundle');

jest.unmock('cosmiconfig');

jest.setTimeout(30 * 1000);

test('require-context/matching.js', async () => {
  await expect(
    execTest('require-context/matching.js'),
  ).resolves.toMatchSnapshot();
});

test('require-context/mode-lazy.js', async () => {
  await expect(
    execTest('require-context/mode-lazy.js'),
  ).resolves.toMatchSnapshot();
});

test('require-context/mode-lazy-once.js', async () => {
  await expect(
    execTest('require-context/mode-lazy-once.js'),
  ).resolves.toMatchSnapshot();
});

test('require-context/mode-eager.js', async () => {
  await expect(
    execTest('require-context/mode-eager.js'),
  ).resolves.toMatchSnapshot();
});

test('require-context/mode-sync.js', async () => {
  await expect(
    execTest('require-context/mode-sync.js'),
  ).resolves.toMatchSnapshot();
});

test('require-context/conflict.js', async () => {
  await expect(
    execTest('require-context/conflict.js'),
  ).resolves.toMatchSnapshot();
});

test('require-context/empty.js', async () => {
  await expect(execTest('require-context/empty.js')).resolves.toMatchSnapshot();
});

test('require-context/empty.js - release', async () => {
  await expect(
    execTest('require-context/empty.js', {dev: false}),
  ).resolves.toMatchSnapshot();
});

async function execTest(entry, {dev = true}: $ReadOnly<{dev: boolean}> = {}) {
  const config = await Metro.loadConfig(
    {
      config: require.resolve('../metro.config.js'),
    },
    {
      transformer: {
        unstable_allowRequireContext: true,
      },
    },
  );

  const result = await Metro.runBuild(config, {
    entry,
    dev,
    minify: !dev,
  });

  return execBundle(result.code);
}
