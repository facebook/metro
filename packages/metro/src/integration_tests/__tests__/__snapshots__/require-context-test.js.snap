// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`require-context/conflict.js 1`] = `
Object {
  "contextModule": Object {
    "./index.js": "contents of subdir-conflict/index.js",
  },
  "normalModule": "contents of subdir-conflict/index.js",
}
`;

exports[`require-context/empty.js - release 1`] = `
Object {
  "error": Object {
    "code": "MODULE_NOT_FOUND",
    "message": "No modules in context",
  },
}
`;

exports[`require-context/empty.js 1`] = `
Object {
  "error": Object {
    "code": "MODULE_NOT_FOUND",
    "message": "No modules in context",
  },
}
`;

exports[`require-context/matching.js 1`] = `
Object {
  "ab": Array [
    "./a.js",
    "./b.js",
  ],
  "abc": Array [
    "./a.js",
    "./b.js",
    "./c.js",
  ],
  "abcd": Array [
    "./a.js",
    "./b.js",
    "./c.js",
    "./nested/d.js",
  ],
}
`;

exports[`require-context/mode-eager.js 1`] = `
Object {
  "./a.js": "a",
  "./b.js": Object {
    "default": "b",
  },
  "./c.js": "c",
  "./nested/d.js": "d",
}
`;

exports[`require-context/mode-lazy.js 1`] = `
Object {
  "./a.js": Object {
    "0": "a",
    "default": "a",
  },
  "./b.js": Object {
    "default": "b",
  },
  "./c.js": Object {
    "0": "c",
    "default": "c",
  },
  "./nested/d.js": Object {
    "0": "d",
    "default": "d",
  },
}
`;

exports[`require-context/mode-lazy-once.js 1`] = `
Object {
  "./a.js": Object {
    "0": "a",
    "default": "a",
  },
  "./b.js": Object {
    "default": "b",
  },
  "./c.js": Object {
    "0": "c",
    "default": "c",
  },
  "./nested/d.js": Object {
    "0": "d",
    "default": "d",
  },
}
`;

exports[`require-context/mode-sync.js 1`] = `
Object {
  "./a.js": "a",
  "./b.js": Object {
    "default": "b",
  },
  "./c.js": "c",
  "./nested/d.js": "d",
}
`;
